src/utils/wait-for-other-promises-to-finish.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+/**
+ * Waits until all other pending promises are processed.
```

Comment:
I know the context of this method because of your explanations, but this comment alone doesn't seem to be enough for external contributors to me. Especially since the method itself does not explicitly wait for other Promises - that is more of how the method is used.

Reply:
> Especially since the method itself does not explicitly wait for other Promises

Actually.. it does. I'm awaiting the fulfilment of the next tick. So yes, this method is waiting. But async.

Reply:
Isn't that more of an implementation detail in the Node engine that it works this way?

Reply:
So what? 
The function name says that it waits until all other promises are processed. After it's done all other promises are processed. I don't see a problem. It does the thing you expect when running it :shrug:

Reply:
Something like that what you wrote above should be included in the ESDoc comment.
> It's awaiting the fulfilment of the next tick. So yes, this method is waiting. But async.

Reply:
*sigh* here you go